### ISSUE TITLE
S2589 FP with 'is' pattern

### ISSUE THREAD

**Issue by andrei-epure-sonarsource** [SonarSource/sonar-dotnet#3123]
### Description

[S2589](https://rules.sonarsource.com/csharp/RSPEC-2589) 

When casting an object to `bool` with `is`, we can easily check if the value is true by chaining the two expressions, like below.

Value can be either true or false, so there should be no S2589 occurrence.

Probably when encountering the `is` expression we are not assigning the correct contraints.

### Repro steps

```
        public bool Foo(object x) => x is bool value && value; // Noncompliant FP S2589
                                                        ^^^^^

        
        public bool Foo(object x) => x is bool value && value == true; // Noncompliant FP S2589

```

### Known workarounds

None

### Related information

* SonarC# Version 8.3

**Comment by andrei-epure-sonarsource** [comment]
maybe related to #2670?

**Comment by odalet** [comment]
I confirm this still exists (using **VS2019 16.6.5 + SonarAnalyzer.CSharp 8.11.0.20529**):

```csharp
var result = value is bool isConnected && isConnected; // False positive S2589 on the last 'isConnected'
```

This code raises S2589...
